subsys/bluetooth/mesh/beacon.c

```diff
@@ -267,14 +283,14 @@ static bool net_beacon_send(struct bt_mesh_subnet *sub, struct bt_mesh_beacon *b
 	if (time_diff < (600 * MSEC_PER_SEC) &&
 		(time_diff < BEACON_THRESHOLD(beacon) ||
 		 time_since_last_recv < (10 * MSEC_PER_SEC))) {
-		return false;
+		return 1;
```

Comment:
It is evaluated in `beacons_send_next`:

```
		if (err < 0) {
			LOG_ERR("Failed to advertise beacon: err %d", err);
			beacon_send_sub_curr = NULL;
			return false;
		}
	} while (err);
```
if err=1 then it will jump to the next subnet

Reply:
It jumps to the next if it is `not zero`. `1` or `-21` doesn't matter :)

Reply:
Ok, so do you want me to return an error code then for this case? It can not be zero.